cf-agent/cf-agent.c

```diff
@@ -546,7 +562,23 @@ static GenericAgentConfig *CheckOpts(int argc, char **argv)
                     exit(EXIT_FAILURE);
                 }
             }
-            break;
+            else if (strcmp(OPTIONS[longopt_idx].name, "show-evaluated-classes") == 0)
+            {
+                if (optarg == NULL)
+                {
+                    optarg = ".*";
+                }
+                config->agent_specific.common.show_classes = xstrdup(optarg);
```

Comment:
There is no need to `strdup()` the `optarg` as it lives for the lifetime of the process.